torch/utils/cpp_extension.py

```diff
@@ -778,7 +799,33 @@ def _check_cuda_version(self):
                     if getattr(cuda_ver, "major", float("nan")) != getattr(torch_cuda_version, "major", float("nan")):
                         raise RuntimeError(CUDA_MISMATCH_MESSAGE.format(cuda_str_version, torch.version.cuda))
                     warnings.warn(CUDA_MISMATCH_WARN.format(cuda_str_version, torch.version.cuda))
-
+                if sys.platform.startswith('linux'):
+                    cuda_compiler_bounds = CUDA_CLANG_VERSIONS if compiler_name.startswith('clang') else CUDA_GCC_VERSIONS
+
+                    min_compiler_version, max_compiler_version = cuda_compiler_bounds[cuda_str_version]
```

Comment:
This would raise an exception if cuda version is not among the list of supported, which would be quite confusing to the developer. Please check if key is present and print a warning instead of hard error here

Reply:
Should the exception raised below be deactivated on this case?